org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanServlet.java

```diff
@@ -58,10 +62,30 @@ public String getName() {
 	public ScanProcess createProcess(ScanBean scanBean, IPublisher<ScanBean> response) throws EventException {
 		
 		if (scanBean.getScanRequest()==null) throw new EventException("The scan must include a request to run something!");
+
+		// Debugging makes code messy but switching this on can prove useful.
+		// Test used because output message does work.
+		debug("Accepting bean", scanBean, response);		
 		preprocess(scanBean);
+		debug("After processing bean (normally no change)", scanBean, response);		
+		
 		return new ScanProcess(scanBean, response, isBlocking());
 	}
 
+	private void debug(String message, ScanBean scanBean, IPublisher<ScanBean> response) {
+		
+		if (!logger.isDebugEnabled()) return;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)